compass/hierarchy.py

```diff
@@ -1,142 +1,40 @@
-        if result_json == {"Message": "Authorization has been denied for this request."}:
-            return [{"id": None, "name": None}]
-
-        result_units = []
-        for unit_dict in result_json:
-            # TODO LiveData check here too
-            parsed = {
-                "id": int(unit_dict["Value"]),
-                "name": unit_dict["Description"],
-                "parent_id": unit_dict["Parent"],
-            }
-
-            if unit_dict["Tag"]:
-                tag = json.loads(unit_dict["Tag"])[0]
-                if tag:
-                    parsed["status"] = tag.get("org_status")
-                    parsed["address"] = tag.get("address")
-                    parsed["member_count"] = tag.get("Members")
-                    parsed["section_type"] = tag.get("SectionTypeDesc")
-
-            result_units.append(parsed)
-
-        return result_units
-
-    def get_members_with_roles_in_unit(self, unit_number):
-        # Construct request data
-
-        dt = datetime.datetime.now()
-        time_uid = f"{dt.hour}{dt.minute}{dt.microsecond // 1000}"
-        data = {"SearchType": "HIERARCHY", "OrganisationNumber": unit_number, "UI": time_uid}
-
-        # Execute search
-        # JSON data MUST be in the rather odd format of {"Key": key, "Value": value} for each (key, value) pair
-        self.post(f"{Settings.base_url}/Search/Members", json=compass_restify(data))
-
-        # Fetch results from Compass
-        search_results = self.get(f"{Settings.base_url}/SearchResults.aspx")
-
-        # Gets the compass form from the returned document
-        form = html.fromstring(search_results.content).forms[0]
-
-        # If the search hasn't worked the form returns an InvalidSearchError - check for this and raise an error if needed
-        if form.action == "./ScoutsPortal.aspx?Invalid=SearchError":
-            raise Exception("Invalid Search")
-
-        # Get the data and return it as a usable python object (list)
-        member_data_string = form.fields["ctl00$plInnerPanel_head$txt_h_Data"] or "[]"
-        member_data = json.loads(member_data_string)
-        for member in member_data:
-            del member["visibility_status"]  # This is meaningless as we can only see Y people
-            del member["address"]  # This doesn't reliably give us postcode and is a lot of data
-            del member["role"]  # This is Primary role and so not useful
-        # member_data = [prop for member in member_data for prop in member if prop not in ["visibility_status", "address", "role"]]
-        return member_data
+from compass._scrapers.hierarchy import CompassHierarchyScraper
+
+level_parent_map = {
+    1: "Organisation",
+    2: "Country",
+    3: "Region",
+    4: "County",
+    5: "District",
+    6: "Group",
+}
+parent_level_map = {v: k for k, v in level_parent_map.items()}
+
+units = {
+    1: "Countries",
+    2: "Regions",
+    3: "Counties",
+    4: "Districts",
+    5: "Groups",
+}
+sections = {
```

Comment:
Understand if you want to be lazy, but these are crying out to be made Enums.

Reply:
xref 5e6b0c8 

This what you were envisaging?